ui/components/app/confirm-page-container/confirm-page-container.component.js

```diff
@@ -140,7 +140,7 @@ const ConfirmPageContainer = (props) => {
       fromAddress,
       global.ethereumProvider,
     );
-    setCollectionBalance(tokenBalance?.balance?.words?.[0] || 0);
+    setCollectionBalance(tokenBalance || 0);
```

Comment:
I wonder how `tokenBalance?.balance?.words?.[0]` became `tokenBalance`. Do you know why? Is it possible that we may want to do something like `tokenBalance || tokenBalance?.balance?.words?.[0] || '0'`?

Reply:
a couple of discoveries:
- fetchTokenBalance calls [balanceOf](https://docs.openzeppelin.com/contracts/3.x/api/token/erc1155#ERC1155-balanceOf-address-uint256-)
- [balanceOf](https://docs.openzeppelin.com/contracts/3.x/api/token/erc1155#ERC1155-balanceOf-address-uint256-) returns uint256

Reply:
There is some recent change to fetchTokenBalance in [this PR](https://github.com/MetaMask/metamask-extension/pull/17891) that could be what changes the structure of returned value. I tried to log the typeOf tokenBalance and the value of tokenBalance itself and I am attaching the results below.

<img width="550" alt="Screenshot 2023-05-11 at 4 45 48 PM" src="https://github.com/MetaMask/metamask-extension/assets/43930900/e4dfc886-b2f7-44d9-b793-9544549b5cb0">

Reply:
Thanks for providing the return value! This is helpful. Looks like it's a BigNumber

ref: https://docs.ethers.org/v5/api/utils/bignumber/

we have `shared/modules/Numeric.ts` which helps us convert values into BigNumbers. Then we have `shared/modules/conversion.utils.ts` which have conversions from BigNumbers to strings. thinking we will want to use a function from the ConversionUtil. I'm unsure why we don't use the [formatUtils method provided in the example](https://docs.ethers.org/v5/api/contract/example/#erc20-meta-methods) (also this [example](https://community.infura.io/t/ethers-js-how-to-retrieve-the-balance-of-an-erc-20-token/6735)). 

I was confused earlier because I was referring to this mock test value in `app/scripts/metamask-controller.test.js`
```
const MOCK_TOKEN_BALANCE = '888';

function MockEthContract() {
  return () => {
    return {
      at: () => {
        return {
          balanceOf: () => MOCK_TOKEN_BALANCE,
        };
      },
    };
  };
}
```

now we know this is a BigNumber object, we can work off of this

Reply:
I confirmed that `.toString` is good to use here
https://docs.ethers.org/v5/api/utils/bignumber/
 
We will want to remove the value `10` we passed to `toString`. Will leave another comment on the updated code. 

Resolving the conversation since it's looking like `tokenBalance?.balance?.words?.[0]` is not needed